### PR TITLE
Porting asynchronous example from NAN repo

### PR DESCRIPTION
**Pull request by NickNaso**
The original example is hosted here:
https://github.com/nodejs/nan/tree/master/examples/async_pi_estimate
Rewrite all the logic using **node-addon-api**
See the following issue
https://github.com/nodejs/abi-stable-node-addon-examples/issues/10

### CODE REVIEW

File: async_pi_estimate/node-addon-api/async.cc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  // It is not safe to access V8, or V8 data structures
+  // here, so everything we need for input and output
+  // should go on `this`.
+  void Execute () {
+    estimate = Estimate(points);
+  }
+
+  // Executed when the async work is complete
+  // this function will be run inside the main event loop
+  // so it is safe to use V8 again
```
Comment (mhdawson) [inline async_pi_estimate/node-addon-api/async.cc]: probably don't want the reference to V8 here.

File: async_pi_estimate/node-addon-api/async.cc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  void OnOK() {
+     Napi::HandleScope scope(Env());
```
Comment (mhdawson) [inline async_pi_estimate/node-addon-api/async.cc]: extra space ?

File: async_pi_estimate/node-addon-api/pi_est.h [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/*********************************************************************
+ * NAN - Native Abstractions for Node.js
+ *
+ * Copyright (c) 2018 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
```
Comment (mhdawson) [inline async_pi_estimate/node-addon-api/pi_est.h]: I don't see this copyright in any of the files being added under nan. I'd expect it to be in both or neither.

File: async_pi_estimate/node-addon-api/async.cc (hunk 1 above)
Comment (NickNaso) [reply async_pi_estimate/node-addon-api/async.cc]: Replaced V8 with JS engine

File: async_pi_estimate/node-addon-api/async.cc (hunk 2 above)
Comment (NickNaso) [reply async_pi_estimate/node-addon-api/async.cc]: Removed space

File: async_pi_estimate/node-addon-api/pi_est.h (hunk 3 above)
Comment (NickNaso) [reply async_pi_estimate/node-addon-api/pi_est.h]: Removed copyright

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by NickNaso** [comment]
Hi @mhdawson I added two examples about event emitter. Could you review the code? Thanks!

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by mhdawson** [comment]
Landed as b6f3e8beed478baac296bb224fa9902d6abad119

**Pull request closed without merging**